thunder/tests/test_networks.py

```diff
@@ -581,4 +581,12 @@ def forward_backward_peak(m, inp):
     mem_thunder = forward_backward_peak(jm, inp)
     mem_eager = forward_backward_peak(m, inp)
 
+    # assert that attention is not automatically recomputed, see
+    # https://github.com/Lightning-AI/lightning-thunder/issues/1646
+    assert not {
+        bsym.sym.name
+        for bsym in thunder.last_backward_traces(jm)[-1].bound_symbols
+        if "attention" in bsym.sym.name and "forward" in bsym.sym.name
+    }
```

Comment:
I think this should also consider `if "sdpa" in bsym.sym.name and "fwd" in bsym.sym.name` as `cudnn_sdpa_fwd` could show up as well because it is one of the default executors if available (or test can just remove this executor while passing to `thunder.jit`).


```python
    # assert that attention is not automatically recomputed, see
    # https://github.com/Lightning-AI/lightning-thunder/issues/1646
    assert not {
        bsym.sym.name
        for bsym in thunder.last_backward_traces(jm)[-1].bound_symbols
        if ("attention" in bsym.sym.name and "forward" in bsym.sym.name) or \
            ("sdpa" in bsym.sym.name and "fwd" in bsym.sym.name)
    }
```

Reply:
Great catch, I switched to `or`ing the attention/sdpa and forward/fwd separately before `and`ing.